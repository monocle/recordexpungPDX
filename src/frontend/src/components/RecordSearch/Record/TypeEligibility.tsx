import React from "react";
import { TypeEligibilityData } from "./types";
import EligibilityReasons from "./EligibilityReasons";

interface Props {
  type_eligibility: TypeEligibilityData;
}

export default function TypeEligibility({ type_eligibility }: Props) {
  return <EligibilityReasons kind="type" eligibility={type_eligibility} />;
}
