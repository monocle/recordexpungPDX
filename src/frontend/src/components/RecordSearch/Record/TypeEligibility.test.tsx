import React from "react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { TypeEligibilityData } from "./types";
import TypeEligibility from "./TypeEligibility";

const typeEligibility = {
  status: "",
  reason: "a good reason",
};

function doRender(type_eligibility: TypeEligibilityData) {
  return renderer
    .create(<TypeEligibility type_eligibility={type_eligibility} />)
    .toJSON();
}

it("renders correctly for an 'Eligible' status", () => {
  const type_eligibility = { ...typeEligibility, status: "Eligible" };
  expect(doRender(type_eligibility)).toMatchSnapshot();
});

it("renders correctly for an 'Ineligible' status", () => {
  const type_eligibility = { ...typeEligibility, status: "Ineligible" };
  expect(doRender(type_eligibility)).toMatchSnapshot();
});

it("renders correctly for an 'Needs More Analysis' status", () => {
  const type_eligibility = {
    ...typeEligibility,
    status: "Needs More Analysis",
  };
  expect(doRender(type_eligibility)).toMatchSnapshot();
});

it("renders correctly for a non specific status", () => {
  const type_eligibility = { ...typeEligibility, status: "not a real status" };
  expect(doRender(type_eligibility)).toMatchSnapshot();
});
