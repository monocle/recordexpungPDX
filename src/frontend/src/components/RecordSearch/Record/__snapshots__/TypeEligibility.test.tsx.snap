// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly for a non specific status 1`] = `"Unknown type eligibility"`;

exports[`renders correctly for an 'Eligible' status 1`] = `
<div
  className="relative mb3 connect connect-type"
>
  <i
    aria-hidden="true"
    className="absolute fas fa-circle gray bg-white outline-2-white z-1"
  />
  <div
    className="ml3 pl1"
  >
    <div
      className=""
    >
      <span
        className="fw7"
      >
        Type: 
      </span>
      a good reason
    </div>
  </div>
</div>
`;

exports[`renders correctly for an 'Ineligible' status 1`] = `
<div
  className="relative mb3 connect connect-type"
>
  <i
    aria-hidden="true"
    className="absolute fas fa-times-circle red bg-white outline-2-white z-1"
  />
  <div
    className="ml3 pl1"
  >
    <div
      className=""
    >
      <span
        className="fw7"
      >
        Type: 
      </span>
      a good reason
    </div>
  </div>
</div>
`;

exports[`renders correctly for an 'Needs More Analysis' status 1`] = `
<div
  className="relative mb3 connect connect-type"
>
  <i
    aria-hidden="true"
    className="absolute fas fa-question-circle purple bg-white outline-2-white z-1"
  />
  <div
    className="ml3 pl1"
  >
    <div
      className=""
    >
      <span
        className="fw7"
      >
        Type: 
      </span>
      a good reason
    </div>
  </div>
</div>
`;
